MatrixKit/Models/Contact/MXKContactManager.m

```diff
@@ -1507,21 +1557,20 @@ - (void)cacheMatrixContacts
     }
 }
 
-- (void)loadCachedMatrixContacts
+- (NSDictionary*)fetchCachedMatrixContacts
 {
     NSDate *startDate = [NSDate date];
     
     NSString *dataFilePath = [self dataFilePathForComponent:matrixContactsFile];
     
     NSFileManager *fileManager = [[NSFileManager alloc] init];
     
-    __block NSDictionary *matrixContactByContactIDCpy = nil;
+    __block NSDictionary *matrixContactByContactID = nil;
     
     if ([fileManager fileExistsAtPath:dataFilePath])
     {
         // Use here processing queue because this queue is used during encoding.
         dispatch_sync(processingQueue, ^{
-            
```

Comment:
The evil `dispatch_sync` is no more required as we assume that this method is now called from a processing queue.
If not, we should make this method async.

Reply:
I agree to get rid of `dispatch_sync` to avoid potential deadlock